js/ui/shellDBus.js

```diff
@@ -516,3 +517,107 @@ var AppStoreService = class {
         this._dbusImpl.emit_signal('ApplicationsChanged', GLib.Variant.new('(as)', [allApps]));
     }
 };
+
+const HackableAppIface = loadInterfaceXML('com.endlessm.HackableApp');
+var HackableApp = class {
+    constructor(manager, session) {
+        this._dbusImpl = Gio.DBusExportedObject.wrapJSObject(HackableAppIface, this);
+        Gio.bus_own_name_on_connection(Gio.DBus.session, 'com.endlessm.HackableApp',
+                                       Gio.BusNameOwnerFlags.REPLACE, null, null);
+
+        this._session = session;
+        this._objectId = manager.getNextId();
+        this._appId = session.app.meta_window.gtk_application_id;
+        this._objectPath = `/com/endlessm/HackableApp/${this._objectId}`;
+
+        // Export DBus Object.
+        this._dbusImpl.export(Gio.DBus.session, this._objectPath);
+
+        this._notifyStateId = session.connect('notify::state',
+                                              this._onSessionStateChanged.bind(this));
+        Main.wm._codeViewManager.connect('session-removed',
+                                         this._onSessionRemoved.bind(this));
```

Comment:
Instead of connecting this here, you could unexport the app directly from the manager. (You can add an `unexport()` method to this class to facilitate that.)